### PR TITLE
enhance Mermaid rendering and styling with theme support

### PR DESCRIPTION
**Pull request by ParkAward**
## Description
This pull request improves the rendering logic of Mermaid diagrams, adds support for dynamic dark and light themes, and applies CSS styling based on the current theme. It also includes logic to wait for Mermaid initialization and the presence of HTML elements with the language-mermaid class.

## Changes

1. Updated Mermaid version from ^9.3.0 to ^10.9.1.
2. Improved Mermaid Rendering Logic:
Added useMermaidEffect hook to initialize Mermaid with either dark or default theme based on Scheme.
Applied animations for smooth transitions when rendering Mermaid diagrams.
3. Added CSS Styling:
Applied theme-based background color to Mermaid code blocks.

## Testing
- The changes were tested focusing on the initial render and reusing Mermaid code inside \<PRE> tags without issues.
- Example: [site](https://sharerice.dev/mermaid-test) (Click! Theme Toggle)

### Related Ticket 
#373

### CODE REVIEW

File: src/routes/Detail/hooks/useMermaidEffect.ts [hunk 1]
```diff
@@ -19,27 +27,53 @@ const waitForMermaid = (interval = 100, timeout = 5000) => {
   })
 }
 const useMermaidEffect = () => {
+  const [memoMermaid, setMemoMermaid] = useState<Map<number, string>>(new Map())
+
+  const { data, isFetched } = useQuery({
```
Comment (ParkAward) [inline src/routes/Detail/hooks/useMermaidEffect.ts:32]: useTheme, useScheme 를 사용하고 싶었지만, cookie로 부터 값을 가져오기 전에
기본 값으로 넣었던 "dark" 로 인해 dark테마만 적용됐어서 useQuery로 작성했습니다.

File: src/routes/Detail/hooks/useMermaidEffect.ts [hunk 2]
```diff
@@ -19,27 +27,53 @@ const waitForMermaid = (interval = 100, timeout = 5000) => {
...
+              )
+              element.innerHTML = svg
+              return
             }
-          )
-        }
+            const svg = await mermaid
+              .render("mermaid" + i, element.textContent || "")
+              .then((res) => res.svg)
+            setMemoMermaid(memoMermaid.set(i, element.textContent ?? ""))
```
Comment (ParkAward) [inline src/routes/Detail/hooks/useMermaidEffect.ts:68]: mermaid.render 동작 후 textContent 값이 svg로 변환됩니다.
렌더 전에 mermaid code로 작성했던 string을 찾을 수 없어
map에 저장 후 사용하도록 구현했습니다.

File: src/routes/Detail/index.tsx [hunk 3]
```diff
@@ -27,4 +27,9 @@ const StyledWrapper = styled.div`
   &[data-type="Paper"] {
     padding: 40px 0;
   }
+  /** Reference: https://github.com/chriskempson/tomorrow-theme **/
```
Comment (ParkAward) [inline src/routes/Detail/index.tsx:30]: 글로벌로 작성된 prism.JS 스타일로 code block의 background 색상이 고정됐습니다.  mermaid code block은 theme와 연결했습니다.

**Review by morethanmin — APPROVED** [review]
확인했습니다. 감사드립니다!

**Pull request merged**